.github/workflows/GenDocs.yml

```diff
@@ -0,0 +1,85 @@
+name: GenDocDeepgramPySDK
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code from main branch        
+        uses: actions/checkout@v3
+        with:
+          ref: main
+          repository: ${{ github.repository }}
+
+      - name: Set up Python  
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pdoc
+          pip install --upgrade py2app
+          pip install python-dotenv
+          pip install -r requirements.txt
+          
+      - name: Get the current tag name
+        run: echo "TAG_NAME=${GITHUB_REF##*/}" >> "$GITHUB_ENV"
+
+      # New Step: Extracting Major Version Number
+      - name: Set the current major version
+        run: echo "MAJOR_VERSION=$(echo $TAG_NAME | cut -d '.' -f 1)" >> "$GITHUB_ENV" # This extracts the major version assuming a 'vX.Y.Z' format
+
+      - name: Generate Documentation for the current tag
+        run: |
+          mkdir -p docs/$MAJOR_VERSION # Note: Changed to use MAJOR_VERSION to create/update the major version folder
+          ROOT_PATH=$(pwd)
+          cd $ROOT_PATH/deepgram
+          pdoc $ROOT_PATH/deepgram -o $ROOT_PATH/docs/$MAJOR_VERSION # Documentation is generated/updated in the major version folder
+
+      - name: Checkout the gh-pages branch
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+          path: gh-pages
+
+      - name: Prepare gh-pages directory structure
+        run: |
+          mkdir -p gh-pages/docs/$MAJOR_VERSION
+      
+      - name: Copy documentation to the gh-pages branch directory
+        run: |
+          cp -r docs/$MAJOR_VERSION gh-pages/docs
```

Comment:
Why not check out first and generate the pages in the correct path to begin with?